atom/browser/api/lib/web-contents.coffee

```diff
@@ -34,6 +34,12 @@ module.exports.wrap = (webContents) ->
     do (name, method) ->
       webContents[name] = -> method.apply controller, arguments
 
+  # Service-worker utilities.
+  webContents.hasServiceWorker = (callback) ->
+    @_hasSW @getUrl(), callback
```

Comment:
We only override APIs in JavaScript when things would be very complicated in C++, for this case we should just call `GetURL` in C++.